__tests__/session.spec.ts

```diff
@@ -0,0 +1,647 @@
+      nextCookies.set(
+        'wos-session',
+        await sealData(mockSession, { password: process.env.WORKOS_COOKIE_PASSWORD as string }),
+      );
+
+      (jwtVerify as jest.Mock).mockImplementation(() => {
+        return true;
+      });
+
+      const request = new NextRequest(new URL('http://example.com'));
+      const result = await updateSession(
+        request,
+        true,
+        {
+          enabled: false,
+          unauthenticatedPaths: [],
+        },
+        process.env.NEXT_PUBLIC_WORKOS_REDIRECT_URI as string,
+        [],
+      );
+
+      expect(result).toBeInstanceOf(NextResponse);
+      expect(result.status).toBe(200);
+      expect(console.log).toHaveBeenCalledWith('Session is valid');
+    });
+
+    it('should attempt to refresh the session when the access token is invalid', async () => {
+      mockSession.accessToken = await generateTestToken({}, true);
+
+      const nextCookies = await cookies();
+      nextCookies.set(
+        'wos-session',
+        await sealData(mockSession, { password: process.env.WORKOS_COOKIE_PASSWORD as string }),
+      );
+
+      (jwtVerify as jest.Mock).mockImplementation(() => {
+        throw new Error('Invalid token');
+      });
+
+      jest.spyOn(workos.userManagement, 'authenticateWithRefreshToken').mockResolvedValue({
+        accessToken: await generateTestToken(),
+        refreshToken: 'new-refresh-token',
+        user: mockSession.user,
+      });
+
+      const request = new NextRequest(new URL('http://example.com'));
+
+      const result = await updateSession(
+        request,
+        true,
+        {
+          enabled: false,
+          unauthenticatedPaths: [],
+        },
+        process.env.NEXT_PUBLIC_WORKOS_REDIRECT_URI as string,
+        [],
+      );
+
+      expect(result.status).toBe(200);
+      expect(console.log).toHaveBeenCalledWith(
+        `Session invalid. Refreshing access token that ends in ${mockSession.accessToken.slice(-10)}`,
+      );
+      expect(console.log).toHaveBeenCalledWith(expect.stringContaining('Refresh successful. New access token ends in'));
+    });
+
+    it('should delete the cookie and redirect when refreshing fails', async () => {
+      jest.spyOn(console, 'log').mockImplementation(() => {});
+
+      mockSession.accessToken = await generateTestToken({}, true);
+
+      const nextCookies = await cookies();
+      nextCookies.set(
+        'wos-session',
+        await sealData(mockSession, { password: process.env.WORKOS_COOKIE_PASSWORD as string }),
+      );
+
+      (jwtVerify as jest.Mock).mockImplementation(() => {
+        throw new Error('Invalid token');
+      });
+
+      jest
+        .spyOn(workos.userManagement, 'authenticateWithRefreshToken')
+        .mockRejectedValue(new Error('Failed to refresh'));
+
+      const request = new NextRequest(new URL('http://example.com'));
+
+      const result = await updateSession(
+        request,
+        true,
+        {
+          enabled: false,
```

Comment:
probably a separate PR but i think this redirect should NOT occur when the middeware auth is not enabled.

Reply:
Yeah I agree, I plan to change that behaviour after this PR lands.

Reply:
FYI: https://github.com/workos/authkit-nextjs/pull/155